modules/apps/29-fee/types/msgs.go

```diff
@@ -1,32 +1,44 @@
 package types
 
-/*
 import (
-	"strings"
-
+	// external library imports
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
-	clienttypes "github.com/cosmos/ibc-go/modules/core/02-client/types"
-	host "github.com/cosmos/ibc-go/modules/core/24-host"
 )
 
-// NewMsg
-func NewMsg() *Msg {
-	return &Msg{
+// msg types
+const (
+	TypeMsgRegisterCounterpartyAddress = "registerCounterpartyAddress"
+)
+
+// NewMsgRegisterCounterpartyAddress creates a new instance of MsgRegisterCounterpartyAddress
+func NewMsgRegisterCounterpartyAddress(sourceAddress, counterpartyAddress string) *MsgRegisterCounterpartyAddress {
+	return &MsgRegisterCounterpartyAddress{
+		Address:             sourceAddress,
+		CounterpartyAddress: counterpartyAddress,
 	}
 }
 
-// ValidateBasic performs a basic check of the Msg fields.
-func (msg Msg) ValidateBasic() error {
+// ValidateBasic performs a basic check of the MsgRegisterCounterpartyAddress fields
+func (msg MsgRegisterCounterpartyAddress) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Address)
+	if err != nil {
+		return sdkerrors.Wrap(err, "failed to convert msg.Address into sdk.AccAddress")
+	}
+
+	_, err = sdk.AccAddressFromBech32(msg.CounterpartyAddress)
+	if err != nil {
+		return sdkerrors.Wrap(err, "failed to convert msg.Address into sdk.AccAddress")
```

Comment:
```suggestion
		return sdkerrors.Wrap(err, "failed to convert counterparty address  into sdk.AccAddress")
```

Reply:
or msg.CounterpartyAddress. Whichever you think is best

Reply:
Good catch. Went with `msg.CounterpartyAddress`